session/session_test.go

```diff
@@ -3048,6 +3048,17 @@ func (s *testSessionSuite2) TestStmtHints(c *C) {
 	c.Assert(tk.Se.GetSessionVars().StmtCtx.GetWarnings(), HasLen, 1)
 	c.Assert(tk.Se.GetSessionVars().GetAllowInSubqToJoinAndAgg(), IsTrue)
 
+	// Test USE_CASCADES hint
+	tk.Se.GetSessionVars().SetEnableCascadesPlanner(true)
+	tk.MustExec("select /*+ USE_CASCADES(false) */ 1;")
+	c.Assert(tk.Se.GetSessionVars().GetEnableCascadesPlanner(), IsFalse)
+	tk.Se.GetSessionVars().SetEnableCascadesPlanner(false)
+	tk.MustExec("select /*+ USE_CASCADES(true) */ 1;")
+	c.Assert(tk.Se.GetSessionVars().GetEnableCascadesPlanner(), IsTrue)
+	tk.MustExec("select /*+ USE_CASCADES(false), USE_CASCADES(true) */ 1;")
+	c.Assert(tk.Se.GetSessionVars().StmtCtx.GetWarnings(), HasLen, 1)
```

Comment:
please also test the warning message.

Reply:
Hi, is it appropriate to write this way?
`c.Assert(tk.Se.GetSessionVars().StmtCtx.GetWarnings()[0].Err.Error(), Equals, "USE_CASCADES() is defined more than once, only the last definition takes effect: USE_CASCADES(true)")`

Reply:
exactly.